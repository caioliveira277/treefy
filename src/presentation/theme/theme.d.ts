import 'styled-components';

interface ThemeColors {
  primary: string;
  secondary: string;
  white: string;
  contrast: string;
  placeholder: string;
  placeholder_light: string;
  rate: string;
  shadow_sm: string;
  border_radius_sm: string;
}

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: ThemeColors;
  }
}
