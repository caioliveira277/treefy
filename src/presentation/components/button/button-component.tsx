import { Button, Container, Text } from './styles';
import { useTheme } from 'styled-components';
import { StyleProp, ViewStyle } from 'react-native';

export type ButtonType = 'default' | 'outline';
export interface ButtonComponentProps {
  children: string;
  style?: StyleProp<ViewStyle>;
  onPress?: () => void;
  type?: ButtonType;
}

export const ButtonComponent: React.FC<ButtonComponentProps> = ({
  children,
  style,
  onPress,
  type = 'default',
}) => {
  const theme = useTheme();

  return (
    <Container style={style}>
      <Button
        testID="button"
        type={type}
        onPress={onPress}
        theme={theme}
        style={{ ...theme.shadows.sm }}
      >
        <Text testID="text" type={type} theme={theme}>
          {children}
        </Text>
      </Button>
    </Container>
  );
};
