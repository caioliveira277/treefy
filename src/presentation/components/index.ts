export * from './button/button-component';
export * from './text-input/text-input-component';
export * from './link/link-component';
export * from './legend/legend-component';
export * from './picker-input/picker-input-component';
export * from './page-loading/page-loading-component';
export * from './toast-message/toast-message-component';
export * from './skeleton-loading/skeleton-loading-component';
export * from './empty-content/empty-content-component';
export * from './search-input/search-input-component';
export * from './animated-height/animated-height';
export * from './loading-overlay/loading-overlay-component';
