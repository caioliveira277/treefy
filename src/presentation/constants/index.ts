export const TREEFY_SUPPORT = {
  email: 'treefy.suporte@gmail.com',
  cel1: '(15) 99782-7880',
  cel2: '(11) 93203-7301',
};
