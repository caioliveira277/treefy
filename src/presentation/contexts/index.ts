export * from './authentication-context';
export * from './introduction-context';
export * from './toast-context';
