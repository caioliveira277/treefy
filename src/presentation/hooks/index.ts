export * from './use-debounce/use-debounce-hook';
