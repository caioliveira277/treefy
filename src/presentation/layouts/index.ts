export * from './public/public-layout';
export * from './profile/profile-layout';
