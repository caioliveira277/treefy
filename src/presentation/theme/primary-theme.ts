import { DefaultTheme } from 'styled-components';

export const primaryTheme: DefaultTheme = {
  colors: {
    primary: '#4ABC86',
    secondary: '#253239',
    white: '#ffffff',
    contrast: '#FCFDFF',
    placeholder: '#C6C6C6',
    placeholder_light: '#EEEEEE',
    rate: '#D56D6D',
    shadow_sm: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    border_radius_sm: '10px',
  },
};
