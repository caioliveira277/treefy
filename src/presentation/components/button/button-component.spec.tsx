import { mockRender } from '@/presentation/test';
import { ButtonComponent, ButtonType } from './button-component';
import faker from '@faker-js/faker';
import { fireEvent, RenderAPI } from '@testing-library/react-native';
import { currentTheme } from '@/presentation/theme';
import 'jest-styled-components/native';

type SutTypes = {
  sut: RenderAPI;
};
type MakeSutProps = {
  text?: string;
  onPress?: () => void;
  type?: ButtonType;
};

const makeSut = ({
  text = 'anyText',
  onPress,
  type,
}: MakeSutProps): SutTypes => {
  const sut = mockRender(
    <ButtonComponent type={type} onPress={onPress}>
      {text}
    </ButtonComponent>
  );
  return {
    sut,
  };
};

describe('ButtonComponent', () => {
  test('Should match with a snapshot', () => {
    const { sut } = makeSut({});
    const { toJSON } = sut;
    expect(toJSON()).toMatchSnapshot();
  });
  test('Should trigger the press event', () => {
    const handlePress = jest.fn();
    const { sut } = makeSut({
      onPress: handlePress,
    });
    const { getByTestId } = sut;
    const button = getByTestId('button');
    fireEvent.press(button);
    expect(handlePress).toHaveBeenCalledTimes(1);
  });
  test('should change the style according to its type', () => {
    const { sut } = makeSut({
      type: 'outline',
    });
    const { getByTestId } = sut;
    const button = getByTestId('button');
    const text = getByTestId('text');

    expect(button).toHaveStyleRule(
      'background-color',
      currentTheme.colors.white
    );
    expect(text).toHaveStyleRule('color', currentTheme.colors.body);
  });
  test('Should match with a snapshot', () => {
    const anyText = faker.datatype.string();
    const { sut } = makeSut({
      text: anyText,
    });
    const { getByTestId } = sut;
    const text = getByTestId('text');
    expect(text.children).toContain(anyText);
  });
});
