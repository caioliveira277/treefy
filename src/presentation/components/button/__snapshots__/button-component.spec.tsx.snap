// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent Should match with a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4ABC86",
        "borderColor": "#4ABC86",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "elevation": 4,
        "justifyContent": "center",
        "minWidth": 200,
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 46,
        "paddingRight": 46,
        "paddingTop": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.23,
        "shadowRadius": 2.62,
      }
    }
    testID="button"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "RobotoSlab_500Medium",
            "fontSize": 16,
          },
        ]
      }
      testID="text"
      theme={
        Object {
          "borders": Object {
            "border_radius_md": "10px",
            "border_radius_sm": "5px",
          },
          "colors": Object {
            "body": "#585960",
            "complete": "#4ABC86",
            "contrast": "#FCFDFF",
            "danger": "#E04244",
            "edit": "#4ABCBC",
            "placeholder": "#C6C6C6",
            "placeholder_light": "#EEEEEE",
            "primary": "#4ABC86",
            "rate_active": "#D56D6D",
            "rate_inactive": "#E5A6A7",
            "secondary": "#253239",
            "sun": "#FEF3A7",
            "water": "#B0D8FE",
            "white": "#ffffff",
          },
          "fonts": Object {
            "families": Object {
              "bold": "RobotoSlab_700Bold",
              "medium": "RobotoSlab_500Medium",
              "regular": "RobotoSlab_400Regular",
            },
            "sizes": Object {
              "lg": "16px",
              "md": "14px",
              "sm": "12px",
              "xl": "18px",
            },
          },
          "shadows": Object {
            "sm": Object {
              "elevation": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2.62,
            },
          },
        }
      }
      type="default"
    >
      anyText
    </Text>
  </View>
</View>
`;
